pkg/autoupdate/tuf/library_lookup.go

```diff
@@ -0,0 +1,75 @@
+package tuf
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/autoupdate"
+)
+
+// CheckOutLatest returns the path to the latest downloaded executable for our binary, as well
+// as its version.
+func CheckOutLatest(binary autoupdatableBinary, rootDirectory string, updateDirectory string, channel string) (string, string, error) {
+	if updateDirectory == "" {
+		updateDirectory = defaultLibraryDirectory(rootDirectory)
+	}
+
+	releasePath, releaseVersion, err := findExecutableFromRelease(binary, LocalTufDirectory(rootDirectory), channel, updateDirectory)
+	if err == nil {
+		return releasePath, releaseVersion, nil
+	}
+
+	// If we can't find the specific release version that we should be on, then just return the executable
+	// with the most recent version in the library
+	return mostRecentVersion(binary, updateDirectory)
+}
+
+// findExecutableFromRelease looks at our local TUF repository to find the release for our
+// given channel. If it's already downloaded, then we return its path and version.
+func findExecutableFromRelease(binary autoupdatableBinary, tufRepositoryLocation string, channel string, baseUpdateDirectory string) (string, string, error) {
+	// Initialize a read-only TUF metadata client to parse the data we already have downloaded about releases.
+	metadataClient, err := readOnlyTufMetadataClient(tufRepositoryLocation)
+	if err != nil {
+		return "", "", errors.New("could not initialize TUF client, cannot find release")
+	}
+
+	// From already-downloaded metadata, look for the release version
+	targets, err := metadataClient.Targets()
+	if err != nil {
+		return "", "", fmt.Errorf("could not get target: %w", err)
+	}
+
+	targetName, _, err := findRelease(binary, targets, channel)
+	if err != nil {
+		return "", "", fmt.Errorf("could not find release: %w", err)
+	}
```

Comment:
I think this is okay, but it feels weird. These are functions that are hard to follow, and work a lot with `github.com/theupdateframework/go-tuf/data`. It feels like we should bake them into something like `metadataClient`. Not sure it's worth it though

Reply:
I think there are a couple places in `autoupdate.go` that feel similar, where it's a lot of work with go-tuf/data. Maybe could be wrapped in a `tufReleaseClient` (or something) that performs the repeated tasks (update from remote TUF repo to download updated targets, find release target among those targets, find what target it's pointing at, return that target). I could do that here, or add it to the TODOs for a subsequent PR if we want to think about it more.

Reply:
I'll leave as-is for now, I guess, and think on it